File: android/app/src/main/java/com/laneshadow/ui/molecules/LSCancelConfirmSheet.kt
package com.laneshadow.ui.molecules

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.Dialog
import androidx.compose.ui.window.DialogProperties
import com.laneshadow.theme.LocalLaneShadowTheme
import com.laneshadow.ui.atoms.ButtonVariant
import com.laneshadow.ui.atoms.ContentColor
import com.laneshadow.ui.atoms.LSButton
import com.laneshadow.ui.atoms.LSText
import com.laneshadow.ui.atoms.TypographyVariant

/**
 * LSCancelConfirmSheet — A centered confirmation sheet for Planning V02.
 *
 * Renders a modal dialog asking the user to confirm cancellation
 * of the current planning operation.
 *
 * @param title Dialog title (e.g., "Cancel this plan?")
 * @param body Dialog body text explaining the consequences
 * @param keepLabel Text for the keep button (default: "Keep thinking")
 * @param cancelLabel Text for the cancel button (default: "Cancel plan")
 * @param onKeep Callback when keep button is tapped
 * @param onCancel Callback when cancel button is tapped
 * @param onDismiss Callback when dialog is dismissed without action
 * @param modifier Modifier for the dialog content
 */
@Composable
fun LSCancelConfirmSheet(
    title: String,
    body: String,
    keepLabel: String = "Keep thinking",
    cancelLabel: String = "Cancel plan",
    onKeep: () -> Unit,
    onCancel: () -> Unit,
    onDismiss: () -> Unit = {},
    modifier: Modifier = Modifier,
) {
    val theme = LocalLaneShadowTheme.current

    Dialog(
        onDismissRequest = onDismiss,
        properties = DialogProperties(
            dismissOnBackPress = true,
            dismissOnClickOutside = true,
            usePlatformDefaultWidth = false,
        ),
    ) {
        // Scrim backdrop - use surface.scrim token (semi-transparent black)
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .background(Color.Black.copy(alpha = 0.4f))
                .padding(theme.space.md),
            horizontalAlignment = androidx.compose.ui.Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center,
        ) {
            // Dialog body - surface.default background
            Column(
                modifier = modifier
                    .background(
                        color = theme.colors.surface.default,
                        shape = RoundedCornerShape(theme.radius.xl),
                    )
                    .padding(theme.space.lg),
                verticalArrangement = Arrangement.spacedBy(theme.space.md),
            ) {
                // Title
                LSText(
                    text = title,
                    variant = TypographyVariant.Opinion.Lg,
                    color = ContentColor.Primary,
                    modifier = Modifier.fillMaxWidth(),
                )

                // Body (italic, secondary color)
                LSText(
                    text = body,
                    variant = TypographyVariant.Opinion.Sm,
                    color = ContentColor.Secondary,
                    modifier = Modifier.fillMaxWidth(),
                )

                // Actions row
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(theme.space.md),
                ) {
                    // Keep button (tertiary - outlined)
                    LSButton(
                        label = keepLabel,
                        variant = ButtonVariant.Tertiary,
                        onClick = onKeep,
                        modifier = Modifier.weight(1f),
                    )

                    // Cancel button (signal - primary filled with signal color)
                    LSButton(
                        label = cancelLabel,
                        variant = ButtonVariant.Primary,
                        onClick = onCancel,
                        modifier = Modifier.weight(1f),
                    )
                }
            }
        }
    }
}

File: ios/LaneShadow/Views/Molecules/LSCancelConfirmSheet.swift
import LaneShadowTheme
import SwiftUI

/// LSCancelConfirmSheet — A centered confirmation sheet for Planning V02.
///
/// Renders a modal dialog asking the user to confirm cancellation
/// of the current planning operation.
public struct LSCancelConfirmSheet: View {
    @Environment(\.theme) private var theme

    private let title: String
    private let body: String
    private let keepLabel: String
    private let cancelLabel: String
    private let onKeep: () -> Void
    private let onCancel: () -> Void

    public init(
        title: String = "Cancel this plan?",
        body: String,
        keepLabel: String = "Keep thinking",
        cancelLabel: String = "Cancel plan",
        onKeep: @escaping () -> Void,
        onCancel: @escaping () -> Void
    ) {
        self.title = title
        self.body = body
        self.keepLabel = keepLabel
        self.cancelLabel = cancelLabel
        self.onKeep = onKeep
        self.onCancel = onCancel
    }

    public var body: some View {
        VStack(spacing: theme.space.md) {
            // Title
            Text(title)
                .font(theme.type.opinion.lg.font)
                .foregroundStyle(theme.colors.content.primary)
                .frame(maxWidth: .infinity, alignment: .leading)

            // Body
            Text(body)
                .font(theme.type.opinion.sm.font)
                .italic()
                .foregroundStyle(theme.colors.content.secondary)
                .frame(maxWidth: .infinity, alignment: .leading)

            // Actions
            HStack(spacing: theme.space.md) {
                // Keep button (tertiary)
                Button(action: onKeep) {
                    Text(keepLabel)
                        .font(theme.type.title.sm.font)
                        .foregroundStyle(theme.colors.content.secondary)
                        .frame(maxWidth: .infinity)
                        .frame(height: 44)
                        .background(
                            RoundedRectangle(cornerRadius: theme.radius.lg)
                                .strokeBorder(
                                    theme.colors.border.default,
                                    lineWidth: theme.borderWidth.sm
                                )
                        )
                }
                .accessibilityIdentifier("cancel-confirm-keep")

                // Cancel button (signal)
                Button(action: onCancel) {
                    Text(cancelLabel)
                        .font(theme.type.title.sm.font)
                        .foregroundStyle(theme.colors.content.primary)
                        .frame(maxWidth: .infinity)
                        .frame(height: 44)
                        .background(
                            RoundedRectangle(cornerRadius: theme.radius.lg)
                                .fill(theme.colors.surface.inset)
                        )
                }
                .accessibilityIdentifier("cancel-confirm-cancel")
            }
        }
        .padding(theme.space.lg)
        .background(
            RoundedRectangle(cornerRadius: theme.radius.xl)
                .fill(theme.colors.surface.card)
        )
        .shadow(radius: theme.elevation.overlay)
        .padding(.horizontal, theme.space.md)
        .accessibilityElement(children: .contain)
        .accessibilityLabel("Cancel planning confirmation")
    }
}

// MARK: - Preview

#Preview {
    ZStack {
        // Simulate scrim background
        Color.black.opacity(0.4)
            .ignoresSafeArea()

        VStack {
            Spacer()

            LSCancelConfirmSheet(
                body: "I've drawn one route already. You can back out now — but I'll toss what I have.",
                onKeep: {},
                onCancel: {}
            )

            Spacer()
                .frame(height: 200) // Simulate bottom sheet positioning
        }
    }
}
